src/Control/Monad/Logic/Sequence/Internal.hs

```diff
@@ -305,6 +325,87 @@ observeAll :: Seq a -> [a]
 observeAll = runIdentity . observeAllT
 {-# INLINE observeAll #-}
 
+-- | Convert @'SeqT' m a@ to @t m a@ when @t@ is some other logic monad
+-- transformer.
+fromSeqT :: (Monad m, Monad (t m), MonadTrans t, Alternative (t m)) => SeqT m a -> t m a
+fromSeqT (toView -> m) = lift m >>= \r -> case r of
+  Empty -> empty
+  a :< s -> pure a <|> fromSeqT s
+
+-- | Convert @'SeqT' m a@ to @'L.LogicT' m a@.
+--
+-- @ toLogicT = 'fromSeqT' @
+toLogicT :: Monad m => SeqT m a -> L.LogicT m a
+toLogicT = fromSeqT
+
+fromLogicT :: Monad m => L.LogicT m a -> SeqT m a
+fromLogicT (L.LogicT f) = fromView $ f (\a v -> return (a :< fromView v)) (return Empty)
+
+instance MFunctor SeqT where
```

Comment:
I'm totally unfamiliar with `MFunctor` so I'm just going to take your word on this stuff.

Reply:
`MFunctor` is pretty simple in practice. It's used to lift monad morphisms (e.g., `lift`, or of course various *isomorphisms* between monads) the way `fmap` is used to lift functions. Have a `SeqT m a` and need a `SeqT (MaybeT m) a`? `hoist lift` will convert it. Etc.